### PR TITLE
Event/Away Site Sprite Spring Cleaning

### PR DESCRIPTION
**Pull request by alsoandanswer**
1. Exorcises fake decal props. Generally horrible, and tends to break after a while (Most of the mapping changes are basically just fixing misaligned sprites and search + replacing fake decal props)
2. Touches up some old event sprites so they can be reused in future.
3. Genericises some Konyang props and assets so they may be used in future events more flexibly.

Sprites that are not mine from Mojave Sun, with some minor modifications.

There's a lot more I would like to finish, but this is all I could muster before the end of the month.
I have further plans to expand upon this for the future Unathi/Skrell/Whatever arcs. Makes my life easier.

### CODE REVIEW

**Comment by alsoandanswer** [comment]
!review

**Comment by DreamySkrell** [comment]
This should be separated into multiple PRs cause it's just touching a ton of things at once, including mapping, on which there are open PRs already - this is going to create merge conflicts for them (or for this PR depending on what is merged first).

**Comment by DreamySkrell** [comment]
image

Looking at mapdiffbot2 renders, the new sprites are kinda. Do not look correct placement-wise. Like the lamp posts look fine as far as the sprite goes I guess, but they're in the middle of the tile now, so they don't allow for 1-tile-wide sidewalks. You would not expect to walk there.

image

Or the chainlink fences look like fine sprites, but the E/W sprite is in the middle of the tile, so it does not connect to N/S fences. So it looks really awkward.

image

Or this here.

**Comment by alsoandanswer** [comment]
> image
> 
> Looking at mapdiffbot2 renders, the new sprites are kinda. Do not look correct placement-wise. Like the lamp posts look fine as far as the sprite goes I guess, but they're in the middle of the tile now, so they don't allow for 1-tile-wide sidewalks. You would not expect to walk there.
> 

I feel that is more of a problem of the map than the sprite itself. Sprites should not occupy intermediaries : it's harder to tell which tile it's on. Also, it adds the issue of pixelshifting everything manually, which is bad.

> image
> 
> Or the chainlink fences look like fine sprites, but the E/W sprite is in the middle of the tile, so it does not connect to N/S fences. So it looks really awkward.
> 

I've also added corner and junction connection sprites to help with this, however, this requires some maps to adjust their fences, since I've changed how they hitbox.

> image
> 
> Or this here.

Again, can be fixed by moving the object.

**Comment by TheDanseMacabre** [comment]
This is such a tremendous visual downgrade that I'm left baffled by this PR. I originally glanced over it and didn't look too deeply into it - the actual meaningful changes to this PR are hardly mentioned in the OP, so I didn't notice them at first. These sprites are from a Fallout server and look extremely ill-fitting, and the idea that it's a good move to delete quality original sprites that haven't even been given a chance to appear ingame to replace them with ill-fitting ports is, and always has been, a very bad idea. I very much hope this PR is closed.

**Comment by DreamySkrell** [comment]
> I feel that is more of a problem of the map than the sprite itself. Sprites should not occupy intermediaries : it's harder to tell which tile it's on. Also, it adds the issue of pixelshifting everything manually, which is bad.

> I've also added corner and junction connection sprites to help with this, however, this requires some maps to adjust their fences, since I've changed how they hitbox.

> Again, can be fixed by moving the object.

That should all be fixed in this PR.

A PR can't just change the functionality of things (like changing hitboxes, or the size/position of the sprite on a turf) without fixing the maps they're used on. Especially since the event arc has already started, it's passed the deadline now, and all the assets should be finished by now.

As for pixelshifting, I think it's fine for decorative things like lamp posts.

**Comment by DreamySkrell** [comment]
image
image

Absolutely against adding real-life american road signs or game consoles. They don't fit on Konyang, or anywhere on the server.

**Comment by alsoandanswer** [comment]
> image image
> 
> Absolutely against adding real-life american road signs or game consoles (XBOX and PlayStation). They don't fit on Konyang, or anywhere on the server.

The highway signs aren't used anywhere on Konyang. They're just there as an additional asset, whosoever wants to use it for an appropriate project. Just because it isn't appropriate *now* as grounds for rejection is counterproductive to building an extensive asset library for future projects.

The game consoles are loose approximations, and they're extremely uncommon props used only a handful of times on the entire server. I really don't think they pose that much of a threat to the suspension of disbelief.

File: code/game/objects/structures/decorative.dm [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/obj/structure/ac_unit
+	name = "AC unit"
+	desc = "A bland air conditioning unit."
+	icon = 'icons/obj/structure/miscellaneous.dmi'
+	icon_state = "ac"
+	layer = 2.7
```
Comment (FluffyGhoster) [inline code/game/objects/structures/decorative.dm:6]:
```suggestion
	layer = BELOW_OBJ_LAYER
```
Probably

File: code/game/objects/structures/decorative.dm [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+	density = 1
```
Comment (FluffyGhoster) [inline code/game/objects/structures/decorative.dm:7]:
```suggestion
	density = TRUE
```

File: code/game/objects/structures/decorative.dm [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+/obj/structure/shelf/Initialize(mapload)
+	. = ..()
+	icon_state = "shelf[pick(1,2,3)]"
+
+/obj/structure/outlet
+	name = "power outlets"
+	desc = "The *other* kind of power-point."
+	icon = 'icons/obj/structure/miscellaneous.dmi'
+	icon_state = "outlet"
+	layer = 2.7
```
Comment (FluffyGhoster) [inline code/game/objects/structures/decorative.dm:53]:
```suggestion
	layer = LAYER_UNDER_TABLE
```

File: code/game/objects/structures/decorative.dm [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+/obj/structure/vent
+	name = "wall vent"
+	desc = "A nondescript vent."
+	icon = 'icons/obj/structure/miscellaneous.dmi'
+	icon_state = "vent"
+	layer = 2.7
```
Comment (FluffyGhoster) [inline code/game/objects/structures/decorative.dm:60]:
```suggestion
	layer = LAYER_UNDER_TABLE
```

File: code/game/objects/structures/decorative.dm [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+// 64x64 stuff
+
+/obj/structure/beach_umbrella
+	name = "overhead beach umbrella"
+	desc = "A tightly secured beach umbrella which looks pretty elegant against the sun."
+	icon = 'icons/obj/structure/64x64_misc.dmi'
+	icon_state = "umbrella"
+	anchored = TRUE
+	layer = 9
```
Comment (FluffyGhoster) [inline code/game/objects/structures/decorative.dm:76]: Pick a define or add one for this

File: code/game/objects/structures/decorative.dm [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+	icon_state = "umbrella_alt"
+
+/obj/structure/crane
+	name = "overhead guiding crane"
+	desc = "A towering crane for lifting industrial sized cargo."
+	icon = 'icons/obj/structure/64x64_machinery.dmi'
+	icon_state = "crane"
+	anchored = TRUE
+	density = TRUE
+	layer = 9
```
Comment (FluffyGhoster) [inline code/game/objects/structures/decorative.dm:88]: As above

File: code/game/objects/structures/light_pole.dm [hunk 7]
```diff
@@ -16,65 +16,83 @@
...
+	light_power = 1.9
 
 /obj/structure/utility_pole/street/classic
+	name = "\improper stone lamp"
+	desc = "A stone lamp commonly found in Konyang."
 	icon_state = "classic_lamp"
 
+/obj/effect/overlay/street_light/classic
+	icon_state = "classic_lamp_light"
+	density = 1
```
Comment (FluffyGhoster) [inline code/game/objects/structures/light_pole.dm:93]:
```suggestion
	density = TRUE
```

**Review by FluffyGhoster — APPROVED** [review]
Code LGTM
For what I'm able to tell, the maps also look good, but I'm not an expert about them

**Comment by DreamySkrell** [comment]
> The interstate signs aren't used anywhere on Konyang. They're just there as an additional asset, whosoever wants to use it for an appropriate project. Same train of thought with genericising things. Just because it isn't appropriate _now_ as grounds for rejection is counterproductive to building an extensive asset library for future projects.
> 
> The game consoles are loose approximations, and they're extremely uncommon props used only a handful of times on the entire server. I really don't think they pose that much of a threat to the suspension of disbelief.

Assets are nice, yes. But I don't think we should be adding stuff just because it may be used some time in the future, maybe years from now. I do not see a place in the lore that could be using modern day american road signs anywhere. Not in Biesel, not even in Sol, except if we literally visited the USA (if it still exists). 

Otherwise, why not just add every asset from every other server? Someone may want to use them 5 or 10 years from now, who knows.

As for the XBOX and Playstation, just, no.

I cannot approve this PR as it is. I do not think American road signs or game consoles are a good addition to the repo.

**Comment by Arrow768** [comment]
@alsoandanswer You have mentioned this in the PR description:

Genericises some Konyang props and assets so they may be used in future events more flexibly.

Could you explain why that is necessary and give a few examples where existing sprites can not be used for other locations.

**Comment by alsoandanswer** [comment]
> @alsoandanswer You have mentioned this in the PR description:
> 
> Genericises some Konyang props and assets so they may be used in future events more flexibly.
> 
> Could you explain why that is necessary and give a few examples where existing sprites can not be used for other locations.

1. Road signs and Traffic Lights/Street Lamps
Namely the stop sign had Korean text on it, which meant it couldn't be used outside of Konyang.
They were also *very* oversaturated, so I had to replace them.
The more muted colors mean they fit into a lot more settings without them glaringly standing out.

2. Signs and whatnot
They use Konyang text, which is fine here. I just adjusted them to use a standardized 64x32 format.
Some props were electronic or holograms, which predisposed them unable to be used in maps/areas without electricty.

3. Hotel Stuff and other fake decal props
They used fake decals which have a tendency to break and be missed in unit tests.
I made them proper objects and did some QC such as standardizing palette and touch-ups.

The rest was mostly just QC.

**Comment by Leroyengloys** [comment]
Synth lore here, it is to my understanding that the PR replaces sprites that were specifically made for the arc, without any consultation from synthlore. Apart from the fact that the ported material is a poor choice in my opinion, I would request that if sprites must absolutely be replaced, that it happens after we are done with Silicon Nightmares.

**Comment by alsoandanswer** [comment]
> Synth lore here, it is to my understanding that the PR replaces sprites that were specifically made for the arc, without any consultation from synthlore. Apart from the fact that the ported material is a poor choice in my opinion, I would request that if sprites must absolutely be replaced, that it happens after we are done with Silicon Nightmares.

It's a wrong understanding. These are just QC changes. They are fundementally unchanged.
Please don't stonewall this any further. The merge conflicts will keep piling up.

**Comment by alsoandanswer** [comment]
Removed the interstate sign and games console references

**Comment by Leroyengloys** [comment]
If you wish to add more signs for more general props, that is fine. But don't replace the existing ones used in Konyang. The blue colour on the arrow sign, the script on the stop sign, are all conscious choices that were made for the konyang aesthetic and I really don't think the quality gets better with the replacements. 

Also we were never informed, so I don't believe voicing the opinion of those most directly affected by it is stonewalling.

**Comment by alsoandanswer** [comment]
> If you wish to add more signs for more general props, that is fine. But don't replace the existing ones used in Konyang. The blue colour on the arrow sign, the script on the stop sign, are all conscious choices that were made for the konyang aesthetic and I really don't think the quality gets better with the replacements.

They have been touched up and readded.

**Review by DreamySkrell — APPROVED** [review]
code and mapping looks fine

**Pull request merged**